src/bin/sol-fbp-generator/main.c

```diff
@@ -580,11 +582,21 @@ generate_node_type_assignments(const struct fbp_data *data)
 static bool
 generate_create_type_function(struct fbp_data *data)
 {
+    char *function;
+
+    if (args.root_type_function)
+        function = strdup(args.root_type_function);
+    else
+        asprintf(&function, "create_%d_%s_type", data->id, data->name);
```

Comment:
Unchecked `asprintf()`: it's undefined what happens with `function` if it returns `-1`.